dbt/adapters/postgres.py

```diff
@@ -23,14 +23,14 @@ def exception_handler(cls, profile, sql, model_name=None,
         except psycopg2.DatabaseError as e:
             logger.debug('Postgres error: {}'.format(str(e)))
 
-            cls.rollback(connection)
+            cls.release_connection(profile, connection_name)
```

Comment:
does this automatically roll back open transactions? would want to be sure we don't create a deadlock because of an open tx

Reply:
yeah, `release_connection` calls `rollback` under the hood if the transaction is open. This change is to fix an issue where we'd try to rollback transactions that weren't open